integration/tests/cook/test_basic.py

```diff
@@ -1879,6 +1879,94 @@ def test_basic_docker_job(self):
         job = util.wait_for_job(self.cook_url, job_uuid, 'completed')
         self.assertIn('success', [i['status'] for i in job['instances']])
 
+    def test_submit_job_with_disk(self):
+        settings_dict = util.settings(self.cook_url)
+        # disk_config_list is a list of regexp's with pool regex, valid disk types, default disk type, and max requestable size of disk
+        disk_config_list = settings_dict.get("pools", {}).get("disk", [])
+        if not disk_config_list:
+            # Submit a job to the default pool specifying request and limit and assert the submission gets silently accepted
+            default_pool = util.default_submit_pool()
+            job_uuid, resp = util.submit_job(
+                self.cook_url,
+                pool=default_pool,
+                disk={'request': 20000.0,
+                      'limit': 30000.0})
+            self.assertEqual(resp.status_code, 201)
+            job = util.load_job(self.cook_url, job_uuid)
+            self.assertEqual(job["disk"]["request"], 20000.0)
+            self.assertEqual(job["disk"]["limit"], 30000.0)
+            self.assertNotIn("type", job["disk"])
+
+            # Submit a job to the default pool specifying disk type and assert the submission gets rejected
+            job_uuid, resp = util.submit_job(
+                self.cook_url,
+                pool=default_pool,
+                disk={'request': 20000.0,
+                      'type': 'pd-ssd'})
+            self.assertEqual(resp.status_code, 400)
+            self.assertTrue("The following disk type is not supported: pd-ssd" in resp.text,
+                            msg=resp.content)
+
+            # Submit a job that specifies disk request greater than max-size on-prem
+            job_uuid, resp = util.submit_job(
+                self.cook_url,
+                pool=default_pool,
+                disk={'request': 200000000.0})
+            self.assertEqual(resp.status_code, 400)
+            self.assertTrue("Disk request specified is greater than max disk size on pool" in resp.text, msg=resp.content)
+
+        else:
+            # Check if there are any active pools
+            active_pools, _ = util.active_pools(self.cook_url)
+            for pool in active_pools:
+                pool_name = pool['name']
+                matching_disk_types = [ii["valid-types"] for ii in disk_config_list if
+                                       re.match(ii["pool-regex"], pool_name)]
+                # If there are no supported disk types for pool, skip test
+                if len(matching_disk_types) == 0 or len(matching_disk_types[0]) == 0:
+                    self.logger.info('There are no disk types configured')
+                    self.skipTest("There are no disk types configured for any active pools")
```

Comment:
I don't think we want to skip inside the `for` loop, right? Because we would be skipping the entire test just because one of the pools didn't have any matching disk types.

Reply:
yes, checked that submission was rejected instead of skipping